packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -51,50 +63,87 @@ contract SuperchainConfig is Initializable, ISemver {
-        paused_ = Storage.getBool(PAUSED_SLOT);
+    /// @notice Extends the pause for a specific identifier by resetting the pause timestamp.
+    /// @param _identifier The address identifier to extend.
+    function extend(address _identifier) external {
+        // Only the Guardian can extend the pause.
+        if (msg.sender != guardian) {
+            revert SuperchainConfig_OnlyGuardian();
+        }
+
+        // Reset the pause timestamp.
+        pauseTimestamps[_identifier] = block.timestamp;
+        emit Paused(LibString.toHexString(_identifier));
     }
 
-    /// @notice Pauses withdrawals.
-    /// @param _identifier (Optional) A string to identify provenance of the pause transaction.
-    function pause(string memory _identifier) external {
-        require(msg.sender == guardian(), "SuperchainConfig: only guardian can pause");
-        _pause(_identifier);
+    /// @notice Checks if the system can be paused for a specific identifier.
+    /// @param _identifier The address identifier to check.
+    /// @return True if the system can be paused for this identifier.
+    function pausable(address _identifier) external view returns (bool) {
+        return pauseTimestamps[_identifier] == 0;
     }
 
-    /// @notice Pauses withdrawals.
-    /// @param _identifier (Optional) A string to identify provenance of the pause transaction.
-    function _pause(string memory _identifier) internal {
-        Storage.setBool(PAUSED_SLOT, true);
-        emit Paused(_identifier);
+    /// @notice Checks if the system is currently paused for a specific identifier.
+    /// @param _identifier The address identifier to check.
+    /// @return True if the system is paused for this identifier and not expired.
+    function paused(address _identifier) public view returns (bool) {
+        uint256 timestamp = pauseTimestamps[_identifier];
+        if (timestamp == 0) return false;
+        return block.timestamp < timestamp + pauseExpiry;
     }
 
-    /// @notice Unpauses withdrawals.
-    function unpause() external {
-        require(msg.sender == guardian(), "SuperchainConfig: only guardian can unpause");
-        Storage.setBool(PAUSED_SLOT, false);
-        emit Unpaused();
+    /// @notice Gets the expiration timestamp for a specific pause identifier.
+    /// @param _identifier The address identifier to check.
+    /// @return The timestamp when the pause expires, or 0 if not paused.
+    function expiration(address _identifier) external view returns (uint256) {
+        uint256 timestamp = pauseTimestamps[_identifier];
+        if (timestamp == 0) return 0;
+        return timestamp + pauseExpiry;
     }
 
     /// @notice Sets the guardian address. This is only callable during initialization, so an upgrade
     ///         will be required to change the guardian.
     /// @param _guardian The new guardian address.
     function _setGuardian(address _guardian) internal {
-        Storage.setAddress(GUARDIAN_SLOT, _guardian);
```

Comment:
If we want to avoid an upgrade function, we would need to read from the guardian slot unless the guardian variable is set

Reply:
Although it would be best if we just added an upgrade function here that converted the guardian slot to the guardian variable